src/infinitecmps/environments.jl

```diff
@@ -121,7 +121,10 @@ function leftenv(H::LocalHamiltonian, Ψρs::InfiniteCMPSData, HL₀=nothing;
         HL₀ = HL₀ - ρL * dot(HL₀, ρR)
     end
     let TL = LeftTransfer(Ψ)
-        HL, infoL = linsolve(hL / norm(hL), HL₀, linalg) do x
+        hLnorm = norm(hL)
+        # rescale to avoid issues with truncate! and linsolve convergence
+        b = (hLnorm < one(hLnorm)) ? hL : hL / hLnorm
+        HL, infoL = linsolve(b, HL₀, linalg) do x
```

Comment:
You will have to restore the norm of `HL` at the end of the process, right? If you did scale `b = hL / hLnorm`, then the solution should afterwise be correced as `HL *= hLnorm`, or thus, in full, line 132 `HL = rmul!(HL + HL', 0.5 * norm(hL))` needs to be replaced with

```julia
HL = rmul!(HL + HL', (hLnorm < one(hLnorm)) ? 0.5 : 0.5*hLnorm)
```

and similar for `HR`.